app/models/service_template_transformation_plan.rb

```diff
@@ -73,11 +78,26 @@ def self.validate_config_info(options)
 
     raise _('Must provide an existing transformation mapping') if mapping.blank?
 
-    vms = if config_info[:vm_ids]
-            VmOrTemplate.find(config_info[:vm_ids])
-          else
-            config_info[:vms]
-          end
+    pre_service_id  = config_info[:pre_service] ? config_info[:pre_service].id : config_info[:pre_service_id]
+    post_service_id = config_info[:post_service] ? config_info[:post_service_id].id : config_info[:post_service_id]
```

Comment:
When do we have the `pre_service` object and when do we only get passed the `pre_service_id`?

Could these lines be simplified to this format:
`pre_service_id  = config_info[:pre_service].try(:id) || config_info[:pre_service_id]`